model_interfaces/interface.py

```diff
@@ -15,14 +15,16 @@ class ChatSession(ABC):
     is_local: bool = False
     max_message_size: int = 1000
 
-    def message_to_agent(self, user_message: str, agent_response: str = "") -> Tuple[str, datetime, datetime]:
+    def message_to_agent(self, user_message: str, agent_response: Optional[str] = None) -> Tuple[str, datetime, datetime]:
         sent_ts = datetime.now()
         old_costs = self.costs_usd
-        response = self.reply(user_message, agent_response)
+        response = self.reply(user_message, agent_response=agent_response)
         reply_ts = datetime.now()
-        assert (
-            self.is_local or old_costs < self.costs_usd
-        ), "The agent implementation is not providing any cost information."
+        # If we are supplying a response from the agent, then don't count costs.
+        if agent_response is not None:
+            assert (
+                self.is_local or old_costs < self.costs_usd
+            ), "The agent implementation is not providing any cost information."
```

Comment:
It's the other way round 🙃 `if agent_response is None:` to run the assert when the agent actually generates an answer.

Reply:
Yeah my testing discovered that :P